### DISCUSSION TITLE
weather submod thingy issue! PLEASE HELP!

### DISCUSSION THREAD

**Discussion by ghost** [Monika-After-Story/MonikaModDev#7937] [general - submod / external content]
i made a api key, and i put the game folder from the submod into autorun, merged it, and monika opens, but when it started up there was a lot of error messages and when i tried to click submods in the menu,it showed an error.

here:

I'm sorry, but an uncaught exception occurred.

While running game code:
  File "renpy/common/00gamemenu.rpy", line 173, in script
    $ ui.interact()
  File "renpy/common/00gamemenu.rpy", line 173, in <module>
    $ ui.interact()
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2631, in execute
    screen submods():
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2631, in execute
    screen submods():
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2634, in execute
    use game_menu(("Submods")):
  File "game/screens.rpy", line 989, in execute
  File "game/screens.rpy", line 989, in execute
  File "game/screens.rpy", line 1010, in execute
  File "game/screens.rpy", line 1013, in execute
  File "game/screens.rpy", line 1019, in execute
  File "game/screens.rpy", line 1022, in execute
  File "game/screens.rpy", line 1051, in execute
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2634, in execute
    use game_menu(("Submods")):
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2638, in execute
    viewport id "scrollme":
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2643, in execute
    vbox:
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2648, in execute
    for submod in sorted(store.mas_submod_utils.submod_map.values(), key=lambda x: x.name):
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2680, in execute
    if submod.settings_pane:
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2681, in execute
    $ renpy.display.screen.use_screen(submod.settings_pane, _name="{0}_{1}".format(submod.author, submod.name))
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2681, in <module>
    $ renpy.display.screen.use_screen(submod.settings_pane, _name="{0}_{1}".format(submod.author, submod.name))
Exception: Screen (u'auto_atmos_change_settings',) is not known.

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "renpy/common/00gamemenu.rpy", line 173, in script
    $ ui.interact()
  File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/ast.py", line 814, in execute
    renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
  File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/python.py", line 1719, in py_exec_bytecode
    exec bytecode in globals, locals
  File "renpy/common/00gamemenu.rpy", line 173, in <module>
    $ ui.interact()
  File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/ui.py", line 285, in interact
    rv = renpy.game.interface.interact(roll_forward=roll_forward, **kwargs)
  File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/core.py", line 2538, in interact
    scene_lists.replace_transient()
  File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/core.py", line 822, in replace_transient
    self.remove(layer, tag)
  File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/core.py", line 1107, in remove
    self.hide_or_replace(layer, remove_index, "hide")
  File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/core.py", line 1031, in hide_or_replace
    d = oldsle.displayable._hide(now - st, now - at, prefix)
  File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/screen.py", line 443, in _hide
    self.update()
  File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/screen.py", line 578, in update
    self.screen.function(**self.scope)
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2631, in execute
    screen submods():
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2631, in execute
    screen submods():
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2634, in execute
    use game_menu(("Submods")):
  File "game/screens.rpy", line 989, in execute
  File "game/screens.rpy", line 989, in execute
  File "game/screens.rpy", line 1010, in execute
  File "game/screens.rpy", line 1013, in execute
  File "game/screens.rpy", line 1019, in execute
  File "game/screens.rpy", line 1022, in execute
  File "game/screens.rpy", line 1051, in execute
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2634, in execute
    use game_menu(("Submods")):
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2638, in execute
    viewport id "scrollme":
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2643, in execute
    vbox:
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2648, in execute
    for submod in sorted(store.mas_submod_utils.submod_map.values(), key=lambda x: x.name):
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2680, in execute
    if submod.settings_pane:
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2681, in execute
    $ renpy.display.screen.use_screen(submod.settings_pane, _name="{0}_{1}".format(submod.author, submod.name))
  File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2681, in <module>
    $ renpy.display.screen.use_screen(submod.settings_pane, _name="{0}_{1}".format(submod.author, submod.name))
  File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/screen.py", line 1145, in use_screen
    raise Exception("Screen %r is not known." % (name,))
Exception: Screen (u'auto_atmos_change_settings',) is not known.

Darwin-20.6.0-x86_64-i386-64bit
Ren'Py 6.99.12.4.2187
Monika After Story 0.12.3

**Comment by ghost** [comment]
> i made a api key, and i put the game folder from the submod into autorun, merged it, and monika opens, but when it started up there was a lot of error messages and when i tried to click submods in the menu,it showed an error.
> 
> here:
> 
> I'm sorry, but an uncaught exception occurred.
> 
> While running game code:
> File "renpy/common/00gamemenu.rpy", line 173, in script
> $ ui.interact()
> File "renpy/common/00gamemenu.rpy", line 173, in
> $ ui.interact()
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2631, in execute
> screen submods():
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2631, in execute
> screen submods():
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2634, in execute
> use game_menu(("Submods")):
> File "game/screens.rpy", line 989, in execute
> File "game/screens.rpy", line 989, in execute
> File "game/screens.rpy", line 1010, in execute
> File "game/screens.rpy", line 1013, in execute
> File "game/screens.rpy", line 1019, in execute
> File "game/screens.rpy", line 1022, in execute
> File "game/screens.rpy", line 1051, in execute
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2634, in execute
> use game_menu(("Submods")):
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2638, in execute
> viewport id "scrollme":
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2643, in execute
> vbox:
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2648, in execute
> for submod in sorted(store.mas_submod_utils.submod_map.values(), key=lambda x: x.name):
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2680, in execute
> if submod.settings_pane:
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2681, in execute
> $ renpy.display.screen.use_screen(submod.settings_pane, _name="{0}_{1}".format(submod.author, submod.name))
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2681, in
> $ renpy.display.screen.use_screen(submod.settings_pane, _name="{0}_{1}".format(submod.author, submod.name))
> Exception: Screen (u'auto_atmos_change_settings',) is not known.
> 
> -- Full Traceback ------------------------------------------------------------
> 
> Full traceback:
> File "renpy/common/00gamemenu.rpy", line 173, in script
> $ ui.interact()
> File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/ast.py", line 814, in execute
> renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
> File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/python.py", line 1719, in py_exec_bytecode
> exec bytecode in globals, locals
> File "renpy/common/00gamemenu.rpy", line 173, in
> $ ui.interact()
> File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/ui.py", line 285, in interact
> rv = renpy.game.interface.interact(roll_forward=roll_forward, **kwargs)
> File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/core.py", line 2538, in interact
> scene_lists.replace_transient()
> File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/core.py", line 822, in replace_transient
> self.remove(layer, tag)
> File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/core.py", line 1107, in remove
> self.hide_or_replace(layer, remove_index, "hide")
> File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/core.py", line 1031, in hide_or_replace
> d = oldsle.displayable._hide(now - st, now - at, prefix)
> File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/screen.py", line 443, in _hide
> self.update()
> File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/screen.py", line 578, in update
> self.screen.function(**self.scope)
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2631, in execute
> screen submods():
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2631, in execute
> screen submods():
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2634, in execute
> use game_menu(("Submods")):
> File "game/screens.rpy", line 989, in execute
> File "game/screens.rpy", line 989, in execute
> File "game/screens.rpy", line 1010, in execute
> File "game/screens.rpy", line 1013, in execute
> File "game/screens.rpy", line 1019, in execute
> File "game/screens.rpy", line 1022, in execute
> File "game/screens.rpy", line 1051, in execute
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2634, in execute
> use game_menu(("Submods")):
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2638, in execute
> viewport id "scrollme":
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2643, in execute
> vbox:
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2648, in execute
> for submod in sorted(store.mas_submod_utils.submod_map.values(), key=lambda x: x.name):
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2680, in execute
> if submod.settings_pane:
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2681, in execute
> $ renpy.display.screen.use_screen(submod.settings_pane, _name="{0}_{1}".format(submod.author, submod.name))
> File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2681, in
> $ renpy.display.screen.use_screen(submod.settings_pane, _name="{0}_{1}".format(submod.author, submod.name))
> File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/screen.py", line 1145, in use_screen
> raise Exception("Screen %r is not known." % (name,))
> Exception: Screen (u'auto_atmos_change_settings',) is not known.
> 
> Darwin-20.6.0-x86_64-i386-64bit
> Ren'Py 6.99.12.4.2187
> Monika After Story 0.12.3

it also said before that about something called _osx_ support

**Comment by ghost** [comment]
> > i made a api key, and i put the game folder from the submod into autorun, merged it, and monika opens, but when it started up there was a lot of error messages and when i tried to click submods in the menu,it showed an error.
> > here:
> > I'm sorry, but an uncaught exception occurred.
> > While running game code:
> > File "renpy/common/00gamemenu.rpy", line 173, in script
> > $ ui.interact()
> > File "renpy/common/00gamemenu.rpy", line 173, in
> > $ ui.interact()
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2631, in execute
> > screen submods():
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2631, in execute
> > screen submods():
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2634, in execute
> > use game_menu(("Submods")):
> > File "game/screens.rpy", line 989, in execute
> > File "game/screens.rpy", line 989, in execute
> > File "game/screens.rpy", line 1010, in execute
> > File "game/screens.rpy", line 1013, in execute
> > File "game/screens.rpy", line 1019, in execute
> > File "game/screens.rpy", line 1022, in execute
> > File "game/screens.rpy", line 1051, in execute
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2634, in execute
> > use game_menu(("Submods")):
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2638, in execute
> > viewport id "scrollme":
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2643, in execute
> > vbox:
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2648, in execute
> > for submod in sorted(store.mas_submod_utils.submod_map.values(), key=lambda x: x.name):
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2680, in execute
> > if submod.settings_pane:
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2681, in execute
> > $ renpy.display.screen.use_screen(submod.settings_pane, _name="{0}_{1}".format(submod.author, submod.name))
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2681, in
> > $ renpy.display.screen.use_screen(submod.settings_pane, _name="{0}_{1}".format(submod.author, submod.name))
> > Exception: Screen (u'auto_atmos_change_settings',) is not known.
> > -- Full Traceback ------------------------------------------------------------
> > Full traceback:
> > File "renpy/common/00gamemenu.rpy", line 173, in script
> > $ ui.interact()
> > File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/ast.py", line 814, in execute
> > renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
> > File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/python.py", line 1719, in py_exec_bytecode
> > exec bytecode in globals, locals
> > File "renpy/common/00gamemenu.rpy", line 173, in
> > $ ui.interact()
> > File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/ui.py", line 285, in interact
> > rv = renpy.game.interface.interact(roll_forward=roll_forward, **kwargs)
> > File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/core.py", line 2538, in interact
> > scene_lists.replace_transient()
> > File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/core.py", line 822, in replace_transient
> > self.remove(layer, tag)
> > File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/core.py", line 1107, in remove
> > self.hide_or_replace(layer, remove_index, "hide")
> > File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/core.py", line 1031, in hide_or_replace
> > d = oldsle.displayable._hide(now - st, now - at, prefix)
> > File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/screen.py", line 443, in _hide
> > self.update()
> > File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/screen.py", line 578, in update
> > self.screen.function(**self.scope)
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2631, in execute
> > screen submods():
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2631, in execute
> > screen submods():
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2634, in execute
> > use game_menu(("Submods")):
> > File "game/screens.rpy", line 989, in execute
> > File "game/screens.rpy", line 989, in execute
> > File "game/screens.rpy", line 1010, in execute
> > File "game/screens.rpy", line 1013, in execute
> > File "game/screens.rpy", line 1019, in execute
> > File "game/screens.rpy", line 1022, in execute
> > File "game/screens.rpy", line 1051, in execute
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2634, in execute
> > use game_menu(("Submods")):
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2638, in execute
> > viewport id "scrollme":
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2643, in execute
> > vbox:
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2648, in execute
> > for submod in sorted(store.mas_submod_utils.submod_map.values(), key=lambda x: x.name):
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2680, in execute
> > if submod.settings_pane:
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2681, in execute
> > $ renpy.display.screen.use_screen(submod.settings_pane, _name="{0}_{1}".format(submod.author, submod.name))
> > File "game/Submod Updater Plugin/submod_updater_plugin.rpy", line 2681, in
> > $ renpy.display.screen.use_screen(submod.settings_pane, _name="{0}_{1}".format(submod.author, submod.name))
> > File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/display/screen.py", line 1145, in use_screen
> > raise Exception("Screen %r is not known." % (name,))
> > Exception: Screen (u'auto_atmos_change_settings',) is not known.
> > Darwin-20.6.0-x86_64-i386-64bit
> > Ren'Py 6.99.12.4.2187
> > Monika After Story 0.12.3
> 
> it also said before that about something called _osx_ support

UGH NOW IT SAYS THIS!!:


I'm sorry, but an uncaught exception occurred.

While running game code:
  File "renpy/common/00start.rpy", line 212, in script call
    call _splashscreen from _call_splashscreen_1
  File "game/splash.rpy", line 165, in script
  File "game/splash.rpy", line 176, in <module>
  File "game/definitions.rpy", line 1075, in validateConditionals
EventException: Failed to evaluate the 'not store.awc_isInvalidAPIKey(store.persistent._awc_API_key)' conditional for the event with the 'awc_monika_player_location' label:
Traceback (most recent call last):
  File "game/definitions.rpy", line 1068, in validateConditionals
  File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/python.py", line 1743, in py_eval_bytecode
    return eval(bytecode, globals, locals)
  File "<none>", line 1, in <module>
AttributeError: 'StoreModule' object has no attribute 'awc_isInvalidAPIKey'
.

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "renpy/common/00start.rpy", line 212, in script call
    call _splashscreen from _call_splashscreen_1
  File "game/splash.rpy", line 165, in script
  File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/ast.py", line 814, in execute
    renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
  File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/python.py", line 1719, in py_exec_bytecode
    exec bytecode in globals, locals
  File "game/splash.rpy", line 176, in <module>
  File "game/definitions.rpy", line 1075, in validateConditionals
EventException: Failed to evaluate the 'not store.awc_isInvalidAPIKey(store.persistent._awc_API_key)' conditional for the event with the 'awc_monika_player_location' label:
Traceback (most recent call last):
  File "game/definitions.rpy", line 1068, in validateConditionals
  File "/Users/jowilson331/Desktop/dokidoki.app/Contents/Resources/autorun/renpy/python.py", line 1743, in py_eval_bytecode
    return eval(bytecode, globals, locals)
  File "<none>", line 1, in <module>
AttributeError: 'StoreModule' object has no attribute 'awc_isInvalidAPIKey'
.

Darwin-20.6.0-x86_64-i386-64bit
Ren'Py 6.99.12.4.2187
Monika After Story 0.12.3

**Comment by multimokia** [comment]
One thing I'll mention, as is, Auto Atmos Change isn't compatible with Mac, normally it _should_ crash due to a missing library for OSX compatibility.

**Comment by multimokia** [comment]
You may want to uninstall it.

**Comment by ghost** [comment]
> You may want to uninstall it.



> One thing I'll mention, as is, Auto Atmos Change isn't compatible with Mac, normally it _should_ crash due to a missing library for OSX compatibility.

ohhhh..ok btw ur really cool can we like be friends?
